src/Sylius/Bundle/ApiBundle/Resources/config/services/api_payment_methods.xml

```diff
@@ -21,5 +21,10 @@
         <service id="Sylius\Bundle\ApiBundle\Provider\CompositePaymentConfigurationProvider">
             <argument type="tagged_iterator" tag="sylius.api.payment_method_handler" />
         </service>
+
+        <service id="Sylius\Bundle\ApiBundle\Provider\CaptureUrlPaymentConfigurationProvider" decorates="Sylius\Bundle\ApiBundle\Provider\CompositePaymentConfigurationProvider">
```

Comment:
No need for decoration, see above.

Reply:
The decoration is here to add the `capture_url` as a default data to the final returned `array`. I wanted to avoid replacing the existing behaviour that replace the entire returned `array`.

As @diimpp already mention it, a better solution would be to create a dedicated endpoint instead of using the existing one.

Reply:
Regarding replacing, there's nothing to replace as there isn't a single provider in the codebase so far. If the idea is that these providers are to be written by downsteam, then I think downstream could as well decorate your provider.

Reply:
And regarding a dedicated endpoint, I'm just not sure if this exact endpoint isn't already the dedicated one.

Reply:
Unfortunately and like you already pointed out on Slack, there is no documentation or sequence to follow. I guess we need some inputs from some Sylius Core team members to choose between creating a new endpoint or using this existing one.

The other reason I made a service is also to be able to easily refactor it in the futur in the case we need to use it in another endpoint.

Reply:
Well, the endpoint is currently a dummy, so I believe it would be a valid approach to default to making it useful :grin:  unless input from Sylius people indicated otherwise.